TestVectors/dafny/TestVectors/src/ParseEsdkJsonManifest.dfy

```diff
@@ -283,6 +233,85 @@ module {:options "-functionSyntax:4"} ParseEsdkJsonManifest {
 
   }
 
+  function V1ToDecryptTestVector(
+    op: EsdkManifestOptions.ManifestOptions,
+    keys: KeyVectors.KeyVectorsClient,
+    name: string,
+    obj: seq<(string, JSON)>,
+    version: SupportedDecryptVersion
+  ) : Result<EsdkDecryptTestVector, string>
+    requires op.Decrypt?
+  {
+    var plaintextLoc :- GetString("plaintext", obj);
+    var ciphertextLoc :- GetString("ciphertext", obj);
+    :- Need(
+         && "file://" < ciphertextLoc
+         && "file://" < plaintextLoc,
+         "Invalid file prefix in test vector"
+       );
+    var masterKeys :- GetArray("master-keys", obj);
+    var keyDescriptions :- GetKeyDescriptions(masterKeys, keys);
+    var keyDescription :- ToMultiKeyDescription(keyDescriptions);
+
+    Success(PositiveV1DecryptTestVector(
+              id := name,
+              version := version,
+              manifestPath := op.manifestPath,
+              ciphertextPath := ciphertextLoc[|FILE_PREPEND|..],
+              plaintextPath := plaintextLoc[|FILE_PREPEND|..],
+              decryptDescriptions := keyDescription,
+              frameLength := None,
+              algorithmSuiteId := None,
+              description :=  name,
+              decryptionMethod := DecryptionMethod.OneShot
+            ))
+  }
+  
+  function V2ToDecryptTestVector(
+    op: EsdkManifestOptions.ManifestOptions,
+    keys: KeyVectors.KeyVectorsClient,
+    name: string,
+    obj: seq<(string, JSON)>,
+    version: SupportedDecryptVersion
+  ) : Result<EsdkDecryptTestVector, string>
+    requires op.Decrypt?
+  {
+    var resultLoc :- GetObject("result", obj);
+    var errorLoc? := GetObject("error", resultLoc);
+    // TODO build negative test vectors appropriately instead of not
+    // accepting any.
+    // Return early if we have a negative test vector
+    if errorLoc?.Success? then
+      Failure(negativeTestVectorFound)
+    else
+
+    var outputLoc :- GetObject("output", resultLoc);
+
+    var plaintextLoc :- GetString("plaintext", outputLoc);
+    var ciphertextLoc :- GetString("ciphertext", obj);
+    :- Need(
+         && "file://" < ciphertextLoc
+         && "file://" < plaintextLoc,
+         "Invalid file prefix in test vector"
+       );
+    var masterKeys :- GetArray("master-keys", obj);
+    var keyDescriptions :- GetKeyDescriptions(masterKeys, keys);
+    var keyDescription :- ToMultiKeyDescription(keyDescriptions);
+
+    Success(PositiveV1DecryptTestVector(
```

Comment:
This test is V2 but it uses V1 vectors? It's probably reasonable but an explanation would be very helpful

Reply:
I was planning on having a PositiveV2DecryptTestVector but figured it was not needed since the positive case is the same for both versions

Reply:
I could name it PositiveLegacyDecryptTestVector?
thoughts?